packages/contracts/contracts/optimistic-ethereum/OVM/execution/OVM_ExecutionManager.sol

```diff
@@ -1900,12 +2102,19 @@ contract OVM_ExecutionManager is iOVM_ExecutionManager, Lib_AddressResolver {
         // Prevent this call from having any effect unless in a custom-set VM frame
         require(msg.sender == address(0));
 
+        // Initialize the EM's internal state, ignoring nuisance gas.
         ovmStateManager = _ovmStateManager;
         _initContext(_transaction);
         messageRecord.nuisanceGasLeft = uint(-1);
 
+        // Increase the caller's balance by whatever amount the eth_call is intending for them to send.
+        uint256 newBalance = ovmBALANCE(_from) + _value;
+        _setOvmEthBalance(_from, newBalance);
```

Comment:
Is this how the EVM works? If you set an arbitrary value it will execute the transaction under the assumption that you have that balance? Can you just double check on remix and/or geth?

Reply:
Okay so I just assumed this was how it worked.  I went and built a little hardhat demo and the reality appears to be even weirder than I thought... I'm gonna leave this one for a separate commit, but check [this bizarre behavior](https://github.com/ben-chain/hardhat-hackathon-boilerplate/pull/1) out if you are interested.

Reply:
Update: this might just be [a bug in hardhat](https://github.com/nomiclabs/hardhat-hackathon-boilerplate/pull/9); will investigate further after addressing other feedback here.

Reply:
Okay, I dug around on this and both infura and remix reject if the value exceeds the account's balance.  Weird to me that this is how the EVM works, but let's stick with it.  Gets rid of the need to have the storage slot anyway!